bin/vyper

```diff
@@ -114,28 +121,27 @@ def exc_handler(contract_name: ContractName, exception: Exception) -> None:
     raise exception
 
 
-def get_interface_codes(codes: ContractCodes) -> Any:
-    interface_codes = {}
+def get_interface_codes(root_path: Path, codes: ContractCodes) -> Any:
+    interface_codes: Dict = {}
     interfaces = {}
 
-    for code in codes.values():
-        interface_codes.update(extract_file_interface_imports(code))
-
-    if interface_codes:
-        for interface_name, interface_path in interface_codes.items():
-            file_path = Path(interface_path.replace('.', '/')).resolve()
-
-            try:
-                suffix = next(i for i in ('.vy', '.json') if file_path.with_suffix(i).exists())
-            except StopIteration:
+    for file_path, code in codes.items():
+        parent_path = root_path.joinpath(file_path).parent
+        base_paths = (parent_path, root_path)
+        new_codes = extract_file_interface_imports(code)
+        for key, value in new_codes.items():
+            interface_path = get_interface_file_path(base_paths, value)
+            if key in interface_codes and interface_codes[key] != interface_path:
                 raise Exception(
-                    f'Imported interface "{interface_path}{{.vy,.json}}" does not exist.'
+                    f"Cannot assign same key to multiple interfaces - '{key}'"
```

Comment:
I actually think the appropriate thing to do here is to have a separate import namespace for each code file.  This is how things work in python and I think we should try to duplicate that behavior.  For example, in python, if different files happened to do `from ... import ... as foo`, there'd be no problem because each file represents a different module with its own namespace.  There's a comment about this [here](https://github.com/ethereum/vyper/pull/1361#issuecomment-474635884) and some discussion.

Reply:
It wouldn't be hard to implement, it just means compiling contracts individually with a fresh call to `get_interface_codes` each time.  @fubuloubu?

Reply:
I like different namespaces better. Ultimately, compatibility at the contract layer relies upon the ABI typing and not having the right text. Being actually strongly typed vs. Python is something to consider.